x-pack/heartbeat/scenarios/framework/framework.go

```diff
@@ -0,0 +1,316 @@
+	sdb.All = append(sdb.All, s...)
+}
+
+func (sdb *ScenarioDB) RunAll(t *testing.T, callback func(*testing.T, *MonitorTestRun, error)) {
+	sdb.RunAllWithATwist(t, nil, callback)
+}
+
+func (sdb *ScenarioDB) RunAllWithATwist(t *testing.T, twist Twist, callback func(*testing.T, *MonitorTestRun, error)) {
+	sdb.Init()
+	for _, s := range sdb.All {
+		s.Run(t, twist, callback)
+	}
+}
+
+func (sdb *ScenarioDB) RunTag(t *testing.T, tagName string, callback func(*testing.T, *MonitorTestRun, error)) {
+	sdb.RunTagWithATwist(t, tagName, nil, callback)
+}
+
+func (sdb *ScenarioDB) RunTagWithATwist(t *testing.T, tagName string, twist Twist, callback func(*testing.T, *MonitorTestRun, error)) {
+	sdb.Init()
+	if len(sdb.ByTag[tagName]) < 1 {
+		require.Failf(t, "no scenarios have tags matching %s", tagName)
+	}
+	for _, s := range sdb.ByTag[tagName] {
+		s.Run(t, twist, callback)
+	}
+}
+
+type MonitorTestRun struct {
+	StdFields stdfields.StdMonitorFields
+	Config    mapstr.M
+	Events    func() []*beat.Event
+	monitor   *monitors.Monitor
+	wait      func()
+	close     func()
+}
+
+func runMonitorOnce(t *testing.T, monitorConfig mapstr.M, location *hbconfig.LocationWithID, stateLoader monitorstate.StateLoader) (mtr *MonitorTestRun, err error) {
+	mtr = &MonitorTestRun{
+		Config:    monitorConfig,
+		StdFields: stdfields.StdMonitorFields{},
+	}
+
+	// make a pipeline
+	pipe := &monitors.MockPipeline{}
+	// pass it to the factory
+	f, sched, closeFactory := setupFactoryAndSched(location, stateLoader)
+	conf, err := config.NewConfigFrom(monitorConfig)
+	require.NoError(t, err)
+	err = conf.Unpack(&mtr.StdFields)
+	require.NoError(t, err)
+
+	mIface, err := f.Create(pipe, conf)
+	require.NoError(t, err)
+	mtr.monitor = mIface.(*monitors.Monitor)
+	require.NotNil(t, mtr.monitor, "could not convert to monitor %v", mIface)
+	mtr.Events = pipe.PublishedEvents
+
+	// start the monitor
+	mtr.monitor.Start()
+	mtr.wait = func() {
+		// wait for the monitor to stop
+		sched.WaitForRunOnce()
+		// stop the monitor itself
+		mtr.monitor.Stop()
+		closeFactory()
+	}
+	mtr.close = closeFactory
+	return mtr, err
+}
+
+func setupFactoryAndSched(location *hbconfig.LocationWithID, stateLoader monitorstate.StateLoader) (factory *monitors.RunnerFactory, sched *scheduler.Scheduler, close func()) {
+	id, _ := uuid.NewV4()
+	eid, _ := uuid.NewV4()
+	info := beat.Info{
+		Beat:            "heartbeat",
+		IndexPrefix:     "heartbeat",
+		Version:         beatversion.GetDefaultVersion(),
+		ElasticLicensed: true,
+		Name:            "heartbeat",
+		Hostname:        "localhost",
+		ID:              id,
+		EphemeralID:     eid,
+		FirstStart:      time.Now(),
+		StartTime:       time.Now(),
+		Monitoring: struct {
+			DefaultUsername string
+		}{
+			DefaultUsername: "test",
+		},
+	}
+
+	sched = scheduler.Create(
+		1,
+		monitoring.NewRegistry(),
+		time.Local,
+		nil,
+		true,
+	)
+
+	if stateLoader == nil {
+		stateLoader = monitorstate.NilStateLoader
```

Comment:
Do we need this explicitly set here? I thought we handle this on our loader interface? fakeloader should have this.

Reply:
yup, will remove, leftover from an old version